src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -258,6 +258,22 @@ public static ImmutableArray<byte> ToImmutable(this MemoryStream stream)
             }
         }
 
+        public static ImmutableArray<TResult> ZipAsArray<T1, T2, TArg, TResult>(this ImmutableArray<T1> self, ImmutableArray<T2> other, TArg arg, Func<T1, T2, TArg, int, TResult> map)
```

Comment:
>this ImmutableArray<T1> self, ImmutableArray<T2> other, TArg arg, Func<T1, T2, TArg, int, TResult> map [](start = 80, length = 102)

Please consider switching the order of `int` and `TArg` parameters to `map`, and the order of `arg` and `map` parameters to method, to match other extension methods:
```
..., Func<T1, T2, int, TArg, TResult> map, TArg arg);
```

Reply:
I'll switch the order in the `map` parameter, but I'm not swapping `map` and `arg`. The existing overloads are painful to use because, when naturally typing, no type information will have been inferred for `TArg` when creating a lambda for `map`. At some point I'd actually like to go change all the rest of them to put `arg` first because it makes intellisense behave significantly better.

---
In reply to: [431355574](https://github.com/dotnet/roslyn/pull/44436#discussion_r431355574) [](ancestors = 431355574)